### PR TITLE
[WIP] Add guides for nixos and nix

### PR DESCRIPTION
**Pull request by ghuntley**
See https://github.com/NixOS/nixpkgs/issues/140344#issuecomment-952478188

### CODE REVIEW

File: docs/guides/nixos/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+```nix
+  virtualisation.oci-containers.containers = {
+     openvscode-server = {
+       image = "gitpod/openvscode-server:latest"; # choose between :latest or :nightly
+       ports = ["3000:3000"];
+       volumes = [
+         "/srv:/home/workspace:cached" # change /srv as appropriate
+       ];
+     };
+  };
```
Comment (bigint) [inline docs/guides/nixos/README.md:20]: Do we need the extra indent at the beginning?

Comment (ghuntley) [reply docs/guides/nixos/README.md:20]: Fixed

File: docs/guides/nix/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
+FROM gitpod/openvscode-server:latest
+
+USER root
+
+# Install Ubuntu packages
+RUN apt-get update \
+    && apt-get install -y curl xz-utils
+
+# Install Nix
+RUN addgroup --system nixbld \
```
Comment (ghuntley) [inline docs/guides/nix/Dockerfile:10]: All references to `vscode-server` will need updating when #86 is resolved

**Comment by dit7ya** [comment]
This is very cool. Looking forward to using Nix as the package manager.

**Review by serenafangirl — APPROVED** [review]

**Comment by bhavitsharma** [comment]
FYI, I've been using this `server.sh` for running on nixos. 
```sh
#!/usr/bin/env nix-shell
#!nix-shell -i bash -p nodejs
#
# Copyright (c) Microsoft Corporation. All rights reserved.
#

case "$1" in
	--inspect*) INSPECT="$1"; shift;;
esac

ROOT="$(dirname "$0")"
PATH=$ROOT/bin:$PATH
node ${INSPECT:-} "$ROOT/out/vs/server/main.js" "$@"
```

**Comment by filiptronicek** [comment]
@ghuntley is this still something you are planning to continue working on 👀? Maybe someone else can continue on it, but I'm not sure how far you got (the markdown looks pretty finished to me).

**Pull request closed without merging**